ethabi/src/token/lenient.rs

```diff
@@ -11,9 +11,15 @@ use crate::{
 	token::{StrictTokenizer, Tokenizer},
 	Uint,
 };
-use regex::Regex;
 use std::borrow::Cow;
 
+macro_rules! regex {
+	($re:literal $(,)?) => {{
+		static RE: once_cell::sync::OnceCell<regex::Regex> = once_cell::sync::OnceCell::new();
+		RE.get_or_init(|| regex::Regex::new($re).unwrap())
+	}};
+}
+
```

Comment:
Think you could also just do:

```suggestion
use once_cell::sync::Lazy;

static RE: Lazy<regex::Regex> = Lazy::new(|| regex::Regex::new(r"([0-9]+\.{0,1}[0-9]+)\s*(ether|gwei|nano|nanoether|wei)").expect("invalid regex"));
```


and just do `RE.matches`

Reply:
instead of doing the macro